src/main/scala/io/iohk/ethereum/ledger/BranchResolution.scala

```diff
@@ -37,52 +38,41 @@ class BranchResolution(blockchain: Blockchain) {
     val oldBlocks = oldBlocksWithCommonPrefix.drop(commonPrefixLength)
     val newHeaders = headersList.drop(commonPrefixLength)
 
-    if (compareByCheckpoints(newHeaders, oldBlocks.map(_.header)))
-      NewBetterBranch(oldBlocks)
-    else
-      NoChainSwitch
-  }
-
-  /**
-    * @return true if newBranch is better than oldBranch
-    */
-  private def compareByCheckpoints(newBranch: Seq[BlockHeader], oldBranch: Seq[BlockHeader]): Boolean =
-    (branchLatestCheckpoint(newBranch), branchLatestCheckpoint(oldBranch)) match {
-      case (Some(newCheckpoint), Some(oldCheckpoint)) =>
-        if (newCheckpoint.number == oldCheckpoint.number)
-          compareByDifficulty(newBranch, oldBranch)
+    val maybeParentWeight: Option[Either[String, ChainWeight]] =
+      oldBlocks.headOption
+        .map(_.header)
+        .orElse(newHeaders.headOption)
+        .map { header =>
+          blockchain
+            .getChainWeightByHash(header.hash)
```

Comment:
Shouldn't we fetch here for the header.parentHash? Syncing to mordor fails if not due to `ChainWeight for 1 not found when resolving branch`